### PR TITLE
Build floodgate against multiple gate versions

### LINKED ISSUE #21: Ability to build Floodgate with different Spinnaker versions.
We should be able to build Floodgate using Gate API for different Spinnaker versions.

### PR DESCRIPTION
**Pull request by boscard**
* Test floodgate changes against latest Gate API version
* 4 times a day test master against all supported Gate API versions
Close #21

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,30 +1,64 @@
 # Golang CircleCI 2.0 configuration file
 #
 # Check https://circleci.com/docs/2.0/language-go/ for more details
-version: 2
+version: 2.1
 jobs:
-  build:
+  generate_swagger:
+    environment:
+      SWAGGER_VERSION=2.4.9
```
Comment (wurbanski) [inline .circleci/config.yml]: I think it's better to use swagger version from file, it's automatically updated when rendering api client locally.

Comment (boscard) [reply .circleci/config.yml]: True, but I don't have this information here because I don't have our repository code here.

Comment (boscard) [reply .circleci/config.yml]: Fixed.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,30 +1,70 @@
...
     docker:
-      - image: circleci/golang:1.14.1
-        environment:
-          GO111MODULE: "on"
-    working_directory: /go/src/github.com/codilime/floodgate
+      - image: circleci/openjdk:11.0.3-jdk-stretch
     steps:
       - checkout
+      - run:
+          name: Prepare extra dirrectories
```
Comment (wurbanski) [inline .circleci/config.yml:16]: typo (`rr`)

**Pull request merged**